tests/e2e/vsan_stretched_cluster_utils.go

```diff
@@ -361,3 +362,141 @@ func createPodsInParallel(client clientset.Interface, namespace string, pvclaims
+	}
+
+	for _, x := range siteHosts {
+		siteEsxMap[x] = true
+	}
+	allMasterIps := getK8sMasterIPs(ctx, client)
+	framework.Logf("all master ips : %v", allMasterIps)
+	sshClientConfig := &ssh.ClientConfig{
+		User: "root",
+		Auth: []ssh.AuthMethod{
+			ssh.Password(k8sVmPasswd),
+		},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+	}
+	framework.Logf("Site esx map : %v", siteEsxMap)
+	// Assuming atleast one master is on that site
+	vcAddress := e2eVSphere.Config.Global.VCenterHostname
+	for _, masterIp := range allMasterIps {
+		cmd := "export GOVC_INSECURE=1;"
+		cmd += fmt.Sprintf("export GOVC_URL='https://administrator@vsphere.local:Admin!23@%s';", vcAddress)
+		cmd += fmt.Sprintf("govc vm.info --vm.ip=%s;", masterIp)
+		result, err := sshExec(sshClientConfig, masterIp, cmd)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		hostIp := strings.Split(result.Stdout, "Host:")
+		host := strings.TrimSpace(hostIp[1])
+		if siteEsxMap[host] {
+			masterIpOnSite = masterIp
+			break
+		}
+	}
+	framework.Logf("Master IP on site : %s", masterIpOnSite)
+	if masterIpOnSite != "" {
+		return masterIpOnSite, nil
+	} else {
+		return "", fmt.Errorf("couldn't find a master running on site")
+	}
+}
+
+// changeLeaderOfContainerToComeUpOnMaster ensures that the leader of a container comes up on
+// a specific master node on that site
+func changeLeaderOfContainerToComeUpOnMaster(ctx context.Context, client clientset.Interface,
+	sshClientConfig *ssh.ClientConfig, csiContainerName string, primarySite bool) error {
+	// Fetch the IP address of master node on that site
+	masterIpOnSite, err := getMasterIpOnSite(ctx, client, primarySite)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	// Get the master Ip where leader of csi container is running
+	allMasterIps := getK8sMasterIPs(ctx, client)
+	// Remove master ip which is on that site from list of master ip of all nodes
+	for i, v := range allMasterIps {
+		if v == masterIpOnSite {
+			allMasterIps = append(allMasterIps[:i], allMasterIps[i+1:]...)
+			break
+		}
+	}
+
+	leaderFoundOnsite := false
+	waitErr := wait.PollImmediate(healthStatusPollInterval, pollTimeout, func() (bool, error) {
+		// Check if leader of csi container comes up on master node of secondary site
+		_, masterIp, err := getK8sMasterNodeIPWhereContainerLeaderIsRunning(ctx, client, sshClientConfig,
+			csiContainerName)
+		framework.Logf("%s container leader is %s on %s ", csiContainerName, masterIp)
+		if err != nil {
+			return false, err
+		}
+		// Pause and kill container of csi container on other master nodes
+		if masterIp == masterIpOnSite {
+			leaderFoundOnsite = true
+			framework.Logf("Leader of %s found on site", csiContainerName)
+			return true, nil
+		}
+
+		csipods, err := client.CoreV1().Pods(csiSystemNamespace).List(ctx, metav1.ListOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		var wg sync.WaitGroup
+		wg.Add(2)
+		go invokeDockerPauseNKillOnContainerInParallel(sshClientConfig, allMasterIps[0],
```

Comment:
just loop through `allMasterIps` and call the go routines, otherwise if we have a setup with more master nodes it will be a problem